connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -530,13 +530,33 @@ private void pauseAll() {
         consumer.pause(consumer.assignment());
     }
 
+    private void put(Collection<SinkRecord> records) {
+        retryWithToleranceOperator.reset();
```

Comment:
this is resetting the operator mid-way, which means it nulls out what record we are processing, so the reports will not have any useful context. for the current implementation here, we want to somehow inform the operator of _all_ the records that are being `put()`.